contracts/MemeFactory.sol

```diff
@@ -418,9 +417,6 @@ abstract contract MemeFactory {
 
         // Check for non-zero token address
         require(memeToken != address(0), "Token creation failed");
-
-        // Update nonce
-        _nonce = localNonce + 1;
```

Comment:
this was very annoying, kept making it hard to reason about

Reply:
need to enhance it a bit more though